File: docs/identities/get-started/_common/code-examples/go/logout_handler.go
package main

import (
	"log"
	"net/http"
)

// LogoutHandler handles the /logout route
func (app *App) logoutHandler(writer http.ResponseWriter, request *http.Request) {
	// Get cookies from the request
	cookies := request.Header.Get("Cookie")

	// Create a logout flow
	logoutFlow, _, err := app.ory.FrontendAPI.CreateBrowserLogoutFlow(request.Context()).
		Cookie(cookies).
		Execute()

	if err != nil {
		log.Printf("Error creating logout flow: %v", err)
		// Redirect to home page if there's an error
		http.Redirect(writer, request, "/", http.StatusSeeOther)
		return
	}

	// Redirect to the logout URL
	http.Redirect(writer, request, logoutFlow.LogoutUrl, http.StatusSeeOther)
}

File: docs/identities/get-started/_common/code-examples/java/LogoutHandler.java
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import sh.ory.ApiException;
import sh.ory.api.FrontendApi;
import sh.ory.model.LogoutFlow;

import java.io.IOException;

@RestController
public class LogoutHandler {
    private final FrontendApi ory;

    public LogoutHandler(FrontendApi ory) {
        this.ory = ory;
    }

    @GetMapping("/logout")
    public void logout(HttpServletRequest request, HttpServletResponse response) throws IOException {
        String cookieHeader = request.getHeader("Cookie");

        try {
            LogoutFlow logoutFlow = ory.createBrowserLogoutFlow(cookieHeader);
            String logoutUrl = logoutFlow.getLogoutUrl();
            if (logoutUrl != null) {
                response.sendRedirect(logoutUrl);
                return;
            }
        } catch (ApiException e) {
            // Error creating logout flow
        }

        // Redirect to home page if there's an error
        response.sendRedirect("/");
    }
}
